File: src/types/cleanup.ts
export type CleanupRisk = "Safe" | "Moderate" | "Expert";

export interface CleanupCategory {
  id: string;
  name: string;
  description: string;
  risk: CleanupRisk;
  fileCount: number;
  totalBytes: number;
  paths: string[];
  enabledByDefault: boolean;
}

export interface CleanupResult {
  categoryId: string;
  filesDeleted: number;
  bytesFreed: number;
  errors: string[];
}

export interface CleanupScanProgress {
  categoryId: string;
  fileCount: number;
  totalBytes: number;
}

export interface CleanupProgress {
  categoryId: string;
  filesDeleted: number;
  bytesFreed: number;
}

File: src-tauri/src/models/cleanup.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CleanupCategory {
    pub id: String,
    pub name: String,
    pub description: String,
    pub risk: CleanupRisk,
    pub file_count: u64,
    pub total_bytes: u64,
    pub paths: Vec<String>,
    pub enabled_by_default: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
pub enum CleanupRisk {
    Safe,
    Moderate,
    Expert,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CleanupResult {
    pub category_id: String,
    pub files_deleted: u64,
    pub bytes_freed: u64,
    pub errors: Vec<String>,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct CleanupScanProgress {
    pub category_id: String,
    pub file_count: u64,
    pub total_bytes: u64,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct CleanupProgress {
    pub category_id: String,
    pub files_deleted: u64,
    pub bytes_freed: u64,
}
